### PR TITLE
Added ADG views, email 

### PR DESCRIPTION
**Pull request by irosenb**

### CODE REVIEW

**Comment by irosenb** [comment]
Not sure why the unicorn timeout commit is showing.

**Comment by irosenb** [comment]
@dinshaw Two things: 

1. I used the `welcome_email` method from the `UserMailer` because it supported any template. Should I rename that method? 

2. Do I need tests for sending the email?

File: app/controllers/adg_registrations_controller.rb [hunk 1]
```diff
@@ -19,6 +19,12 @@ def create
       adg_answer.save!
     end
     current_user.update_attribute(:adg_preference_ids, params[:adg_registration][:adg_preference_ids])
+
+    adg_template = CmsPage.where(reference_string: 'Email::Adg').first    
+    if adg_template
+      UserMailer.welcome_email(current_user, adg_template).deliver
+    end
+
```
Comment (dinshaw) [inline app/controllers/adg_registrations_controller.rb]: I'd prefer to see this mailer logic tucked away somewhere like a UserNotifier class, or somesuch
```
...
UserNotifier.send_adg_email
UserNotifier.send_welcome_email
...

Prob not a great idea to hack the welcome_email method for the adg one. But if you need to do some refactoring to dry things up, by all means do.

**Comment by dinshaw** [comment]
As for testing mailers, we do had something in place for that, but its not mandatory.
But... make sure the template parsing is covered - so introduction of a syntax error would fail - and that the logic around it is tested.

**Comment by irosenb** [comment]
How does the templating for the CMS email pages work? Is there any?

**Comment by irosenb** [comment]
Also @dinshaw what would constitute a "syntax error"?

**Comment by irosenb** [comment]
Okay @dinshaw made a few changes. 

Couldn't really abstract much. Everything is entangled (`@body, @user`, etc).

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -5,7 +5,17 @@ class UserMailer < ActionMailer::Base
...
+    @body = welcome_email.body
+    subject = welcome_email.subject
+    user.attributes.keys.each do |key|
+      @body.gsub!("@#{key.to_s}", user[key].to_s)
+      subject.gsub!("@#{key.to_s}", user[key].to_s)
+    end
+    mail(to: user.email, subject: subject)
+  end
+
+  def adg_email(user, welcome_email)
```
Comment (dinshaw) [inline app/mailers/user_mailer.rb]: Shouldn't read `welcome_email` any more.

**Comment by dinshaw** [comment]
:+1:

**Pull request merged**